applications/CompressiblePotentialFlowApplication/custom_elements/incompressible_perturbation_potential_flow_element.cpp

```diff
@@ -454,13 +454,12 @@ void IncompressiblePerturbationPotentialFlowElement<Dim, NumNodes>::CalculateLef
     // Calculate shape functions
     GeometryUtils::CalculateGeometryData(GetGeometry(), data.DN_DX, data.N, data.vol);
 
-    const double free_stream_density = rCurrentProcessInfo[FREE_STREAM_DENSITY];
-
-    GetWakeDistances(data.distances);
-
     BoundedMatrix<double, NumNodes, NumNodes> lhs_total = ZeroMatrix(NumNodes, NumNodes);
+    BoundedMatrix<double, NumNodes, NumNodes> lhs_wake_condition = ZeroMatrix(NumNodes, NumNodes);
 
-    ComputeLHSGaussPointContribution(data.vol*free_stream_density, lhs_total, data);
+    CalculateBlockLeftHandSideWakeElement(lhs_total, lhs_wake_condition, data, rCurrentProcessInfo);
+
+    GetWakeDistances(data.distances);
```

Comment:
Use utilities here
```suggestion
    PotentialFlowUtilities::GetWakeDistances(data.distances);
```

Reply:
You are right, I will remove this function completely from here